Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -3714,16 +3723,85 @@ void Document::append_pending_animation_event(Web::DOM::Document::PendingAnimati
     m_pending_animation_event_queue.append(event);
 }
 
-void Document::element_id_changed(Badge<DOM::Element>)
+static bool is_potentially_named_element(DOM::Element const& element)
+{
+    return is<HTML::HTMLEmbedElement>(element) || is<HTML::HTMLFormElement>(element) || is<HTML::HTMLIFrameElement>(element) || is<HTML::HTMLImageElement>(element) || is<HTML::HTMLObjectElement>(element);
+}
+
+static bool is_potentially_named_element_by_id(DOM::Element const& element)
+{
+    return is<HTML::HTMLObjectElement>(element) || is<HTML::HTMLImageElement>(element);
+}
```

Comment:
Are there spec links we can reference for this list of elements? Or is it just scattered throughout the spec?